R/mainTests.r

```diff
@@ -2556,36 +2556,39 @@ indicator_dp_completed <- function(meta) {
   return(output)
 }
 
-ethnicity_headers <- function(meta) {
-  # First find any ethnicity type columns that don't have the standard col_names
-  ethnicity_standard_headers <- c("ethnicity_major", "ethnicity_minor", "ethnicity_detailed", "minority_ethnic")
-  ethnicity_columns <- meta %>%
-    filter(
-      grepl("ethnic", tolower(col_name)),
-      !(col_name %in% ethnicity_standard_headers)
-    ) %>%
-    pull(col_name)
-  if (length(ethnicity_columns) == 0) {
+standard_filter_headers <- function(meta) {
+  search_strings <- harmonised_col_names %>%
+    pull(col_name_search_string) %>%
```

Comment:
This wasn't the way I was expecting you to do it - my first assumption was it would work like the [geographic_catch()](https://github.com/dfe-analytical-services/dfe-published-data-qa/blob/a9570b1248b64f6eb52c3be6f51e92f6e754be60/R/mainTests.r#L2008) with a [regex for key phrases in columns](https://github.com/dfe-analytical-services/dfe-published-data-qa/blob/a9570b1248b64f6eb52c3be6f51e92f6e754be60/R/knownVariables.r#L113)

What you have seems to work fine for the list of aliases we have now though!

Reply:
I've made a few small changes now removing the for loop and incorporating filter_grouping_column entries into the check. It's a little more like `geographic_catch()`, but probably still not quite what you were expecting!